packages/extension-browser/src/content-script/connector.ts

```diff
@@ -147,7 +161,7 @@ export default class WebExtensionConnector implements IConnector {
         this.setFetchElement(event);
         const type = this.setFetchType(event);
 
-        await this._engine.emitAsync(`fetch::end::${type}` as 'fetch::end::*', event);
+        await this.handleFetch(`fetch::end::${type}` as 'fetch::end::*', { fetchEnd: event });
```

Comment:
Thinking about this some more I realized it only fixes the Chrome code path for the `Fetcher` class. When running in Firefox we return the response of a `fetch` call initiated by `Fetcher` directly. This means `setFetchElement` and `setFetchType` never get a chance to run in that context so we're still missing `element` and `mediaType` data.

Reply:
Arguably `element` isn't as relevant here since these were triggered by a `fetchContent` call to begin with. This means the caller should have the `element` already so we don't need to wait for it here. And `mediaType` is derived from the headers so we can call that anytime.

TL;DR - I think we can roll back this particular change, add the `mediaType` from within the `Fetcher` class itself, and require callers of `fetchContent` to set the `element`.

Reply:
Updated. The `element` is already added in `parser-manifest` so no need to re-add it.